plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/editor/hyperlinking/packagejson/PackageJsonHyperlinkHelperExtension.java

```diff
@@ -0,0 +1,259 @@
+/**
+ * Copyright (c) 2018 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.ui.editor.hyperlinking.packagejson;
+
+import java.nio.file.Path;
+import java.util.List;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.jface.text.Region;
+import org.eclipse.jface.text.hyperlink.IHyperlink;
+import org.eclipse.n4js.N4JSGlobals;
+import org.eclipse.n4js.json.JSON.JSONPackage;
+import org.eclipse.n4js.json.JSON.JSONStringLiteral;
+import org.eclipse.n4js.json.JSON.NameValuePair;
+import org.eclipse.n4js.json.ui.editor.hyperlinking.HyperlinkHelperExtension;
+import org.eclipse.n4js.packagejson.PackageJsonProperties;
+import org.eclipse.n4js.projectModel.IN4JSProject;
+import org.eclipse.n4js.resource.XpectAwareFileExtensionCalculator;
+import org.eclipse.n4js.ui.internal.N4JSEclipseModel;
+import org.eclipse.n4js.ui.internal.N4JSEclipseProject;
+import org.eclipse.n4js.ui.projectModel.IN4JSEclipseSourceContainer;
+import org.eclipse.n4js.utils.ProjectDescriptionUtils;
+import org.eclipse.xtext.naming.QualifiedName;
+import org.eclipse.xtext.nodemodel.INode;
+import org.eclipse.xtext.nodemodel.util.NodeModelUtils;
+import org.eclipse.xtext.resource.EObjectAtOffsetHelper;
+import org.eclipse.xtext.resource.XtextResource;
+import org.eclipse.xtext.ui.editor.hyperlinking.XtextHyperlink;
+import org.eclipse.xtext.util.Pair;
+import org.eclipse.xtext.util.Tuples;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Strings;
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+
+/**
+ *
+ */
+public class PackageJsonHyperlinkHelperExtension implements HyperlinkHelperExtension {
```

Comment:
We need a naming convention to distinguish between the Hyperlink implementations that can be registered via extension point and the singleton n4js hyperlink implementation that hooks into Xtext hyperlink support. This applies to Proposal.

Reply:
Also decide for either Json or JSON in the naming convention